08-components-lab/src/01-timeconverter.jsx

```diff
@@ -8,5 +8,49 @@
 
 import React from 'react';
 
-export default React.createClass({
-});
+export default class TimeConverter extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            number : 0
+        }
+    }
+
+    static propTypes = {
+        number: React.propTypes.number.isRequired
+    };
+
+    getSeconds(){
+        return Number(this.state.number);
+    };
+    getMinutes(){
+        return Number(this.state.number / 60);
+    };
+    getHours(){
+        return Number(this.state.number / 3600);
+    }
+
+    calc = (evt) => {
```

Comment:
👍 for using arrow functions to bind methods properly. Another solution is to bind them in constructor. The benefit of binding methods in constructor is that you will get more descriptive stack trace when debugging.